src/posts/posts.service.ts

```diff
@@ -14,13 +14,45 @@ export class PostsService {
   /**
    * @description find all posts
    */
-  async getAllPosts(): Promise<Array<PostEntity>> {
+  async getAllPosts(
+    authorId?: string,
+    hashtags?: string[] | null,
+  ): Promise<Array<PostEntity>> {
     // TODO: implementation pagination (size + limit)
-    // TODO: implement filter by author
     // TODO: implement filter by hashtag
-    return this.postsRepository.find({
-      take: 100,
-      order: { createdAt: 'DESC' },
+    const queryBuilder = this.postsRepository.createQueryBuilder('posts');
+
+    if (authorId) {
+      queryBuilder
+        .leftJoinAndSelect('posts.author', 'author')
+        .where(`posts.author = :authorId`, { authorId })
+        .addSelect('posts.created_at');
+    }
+
+    if (hashtags && hashtags.length > 0) {
+      // TODO
+    }
+
+    return authorId
+      ? queryBuilder.orderBy('posts.created_at', 'DESC').limit(100).getMany()
+      : this.postsRepository.find({
```

Comment:
If we have created a queryBuilder, we can use the same query builder here no?

`queryBuilder.orderBy('posts.created_at', 'DESC').limit(100).getMany()` this line should work right for both cases?

Reply:
Yes, I think we can. In fact, previously, I tried using only `queryBuilder.addSelect('posts.created_at').orderBy('posts.created_at', 'DESC').limit(100).getMany();`.

But, the problem was `queryBuilder` didn't seem to eager-load all the relational entities unless `leftJoinAndSelect()` was applied, whereas `find` was eager loading all the relational entities by default. (Reference: [Eager and Lazy relations](https://orkhan.gitbook.io/typeorm/docs/eager-and-lazy-relations))

That's why I had to apply conditional logic to check whether there are any query params in the request.

But now, I've made some modifications while creating the `queryBuilder`, so that it can support both cases. Let me know what you think of it. :) <br> <br>

***After Modifications***
- [x] Get all posts
![get-all-posts](https://user-images.githubusercontent.com/39665013/126995382-fb630407-2494-4467-a43b-1f513c6a2347.png)

- [x] Get all posts filtered by author
![get-all-posts-filtered-by-author](https://user-images.githubusercontent.com/39665013/126995477-197f3e0f-a632-4ec9-a343-79a7af0a3bb9.png)